bpf/tests/bpf_nat_icmp6.h

```diff
@@ -0,0 +1,366 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#pragma once
+
+#include <bpf/ctx/skb.h>
+#include <bpf/api.h>
+#include "common.h"
+#include "pktgen.h"
+
+#define ENABLE_IPV4
+#define ENABLE_SCTP
+#define ENABLE_IPV6
+#define ENABLE_NODEPORT
+#define ENABLE_MASQUERADE_IPV6
+
+#define NODE_ONE { .addr = v6_node_one_addr }
+#define EXT_IP { .addr = v6_ext_node_one_addr }
+#define POD_IP { .addr = v6_pod_one_addr }
+
+#define FROM_NETDEV	0
+#define TO_NETDEV	1
+
+/*
+ * Input packet represents a device sending a PKT_TOO_BIG response ICMPv6
+ * message due to a MTU pathing issue following a TCP being sent to a l4
+ * address tuple.
+ * Note: a similar remapping will be applied to other supported l4 protocols.
+ *
+ * ┌────────────────────────────────┐
+ * │  L2 Header                     │
+ * ├────────────────────────────────┤
+ * │  IPV6 Header:                  │
+ * │    saddr: Remote Endpoint IP   │
+ * │    daddr: Cilium Node IP       │
+ * ├────────────────────────────────┤
+ * │  ICMPv6 Header:                │
+ * │    type: PKT_TOO_BIG (2)       │
+ * ├────────────────────────────────┤
+ * │  IPV6 Header (Inner):          │
+ * │    saddr: Cilium Node IP       │
+ * │    daddr: Remote Endpoint IP   │
+ * ├────────────────────────────────┤
+ * │  TCP Header:                   │
+ * │  ...                           │
+ * └────────────────────────────────┘
+ *
+ * Following SNAT6, it should be remapped as follows:
+ *
+ * ┌────────────────────────────────┐
+ * │  L2 Header                     │
+ * ├────────────────────────────────┤
+ * │  IPV6 Header:                  │
+ * │    saddr: Cilium Node IP       │
+ * │    daddr: Pod Endpoint IP      │
+ * ├────────────────────────────────┤
+ * │  ICMPv6 Header:                │
+ * │    type: PKT_TOO_BIG (2)       │
+ * ├────────────────────────────────┤
+ * │  IPV6 Header (Inner):          │
+ * │    saddr: Cilium Node IP       │
+ * │    daddr: Remote Endpoint IP   │
+ * ├────────────────────────────────┤
+ * │  TCP Header:                   │
+ * │  ...                           │
+ * └────────────────────────────────┘
+ *
+ * Ref: https://datatracker.ietf.org/doc/html/rfc4443#section-3.2
+ */
+__always_inline int gen_pmtu_pkt(struct pktgen *builder, int l4_type)
+{
+	struct ethhdr *l2;
```

Comment:
Maybe you want to define all these as NIL. Per the C spec, these stack vars may contain garbage.